compendium/DeclarativeServices/src/SCRBundleExtension.cpp

```diff
@@ -61,10 +67,15 @@ SCRBundleExtension::SCRBundleExtension(
     metadataparser->ParseAndGetComponentsMetadata(scrMetadata);
   for (auto& oneCompMetadata : componentsMetadata) {
     try {
-      auto compManager = std::make_shared<ComponentManagerImpl>(
-        oneCompMetadata, registry, bundleContext, logger, threadpool);
+      auto compManager = std::make_shared<ComponentManagerImpl>(oneCompMetadata,
+                                                                registry,
+                                                                bundleContext,
+                                                                logger,
+                                                                threadpool,
```

Comment:
Patty: check to see if "threadpool" needs to be included here

Reply:
Yes it does. threadpool has to be passed to the ComponentConfigurationImpl object so that when a factory instance is created it can be passed to the ComponentManagerImpl constructor.

Reply:
The only classes that use threadpool are SCRBundleExtension, ComponentManagerImpl and ConfigurationNotifier. If I move threadpool into ConfigurationNotifier I can take it out of the rest of the classes (ComponentConfigurationImpl, ComponentConfigurationFactory, CMEnabledState, etc) that don't use it.